R/table.R

```diff
@@ -0,0 +1,105 @@
+#' Generics for tabular vectors
+#'
+#' @description
+#'
+#' \Sexpr[results=rd, stage=render]{vctrs:::lifecycle("experimental")}
+#'
+#' These generics make it possible to work with tables, i.e. vectors
```

Comment:
What calling them rectangular vectors? To me, that more clearly implies that each element is the same size.

Reply:
I'm not worried about confusion with tibbles, nor with data tables. A table is a generic term. 

So far I've enjoyed writing about "tabular" vectors to mean vectors with two directions. I don't think I would like "rectangular" as much as it's a long word, but if you prefer rectangular we can go for it. Regarding the prefix, `rect_` has the downside of being 4 letters, and `rec_` might evoke records, which is a different (though related) concept.

Reply:
I think the main problem of "rectangular" is that it is a shape qualifier, thus it only makes sense as a qualifier of vector. In that regard I prefer "tabular" because it makes it natural to talk about tables as nouns, to collectively refer to matrices and data frames.

On the other hand it's not clear yet that we want to handle matrices as part of this API, currently they are an error. Maybe the API should be purely about data frames, and in this case maybe `df_` is better.

It would be too much work to rename in this PR as other branches are depending on it, so we still have time to think about this.

Reply:
The main problem is that `tbl` is the super class that tibbles use — I think the chance of confusion are high. And "table" already has existing meaning and class (`table()`)

(If I succeed in convincing you that rectangular or rectangle is a good name, I'd suggest `rct` as a prefix)

Reply:
ok, I'll change to rectangle in an ulterior PR.